sensors/triggers/params.go

```diff
@@ -272,12 +292,18 @@ func getValueWithTemplate(value []byte, templString string) (string, error) {
 	return out, nil
 }
 
-// getValueByKey will return the value in the raw json bytes at the provided key,
+// getValueByKey will return the value as raw json or a string and value's type at the provided key,
+// Value type (jsonType or stringType or errorType). JSON represent a block while String represent a single value.
 // or an error if it does not exist.
-func getValueByKey(value []byte, key string) (string, error) {
+func getValueByKey(value []byte, key string) (string, string, error) {
 	res := gjson.GetBytes(value, key)
 	if res.Exists() {
-		return res.String(), nil
+		if res.Type.String() == stringType {
+			return res.String(), res.Type.String(), nil
+		} else if res.Type.String() == jsonType {
+			return res.Raw, res.Type.String(), nil
+		}
+		return res.Raw, res.Type.String(), nil
 	}
-	return "", fmt.Errorf("key %s does not exist to in the event payload", key)
+	return "", errorType, fmt.Errorf("key %s does not exist to in the event payload", key)
```

Comment:
no need to have an `errorType`, same reason as above.

Reply:
removed